### ISSUE TITLE
LazyModule support

### ISSUE THREAD

**Issue by anijain2305** [pytorch/torchdynamo#299]
@jansel Should we support LazyModules? They are experimental.

~~~
import torch
import torchdynamo


def fn(lazy_cls, cls):
    lazy_module = lazy_cls(affine=False, track_running_stats=False)
    input = torch.ones(16, 16, 16)
    out = lazy_module(input)
    assert isinstance(lazy_module, cls)

fn(torch.nn.LazyBatchNorm1d, torch.nn.BatchNorm1d)

torchdynamo.config.debug = True
torchdynamo.config.trace = True
print(" -------------------------------------------")
with torchdynamo.optimize("eager"):
    fn(torch.nn.LazyBatchNorm1d, torch.nn.BatchNorm1d)
~~~

When a lazy module is called, the class type changes. 

These are the failures from PyTorch tests.

~~~
[gw1] FAILED test/test_nn.py::TestLazyModules::test_lazy_batchnorm2d_pickle
[gw74] FAILED test/test_nn.py::TestLazyModules::test_lazy_instancenorm3d_pickle
[gw93] FAILED test/test_nn.py::TestLazyModules::test_lazy_instancenorm2d_pickle
[gw82] FAILED test/test_nn.py::TestLazyModules::test_lazy_instancenorm1d
[gw85] FAILED test/test_nn.py::TestLazyModules::test_lazy_batchnorm1d
[gw82] FAILED test/test_nn.py::TestLazyModules::test_lazy_instancenorm1d_pickle
[gw1] FAILED test/test_nn.py::TestLazyModules::test_lazy_batchnorm3d
[gw93] FAILED test/test_nn.py::TestLazyModules::test_lazy_instancenorm3d
[gw85] FAILED test/test_nn.py::TestLazyModules::test_lazy_batchnorm1d_pickle
[gw1] FAILED test/test_nn.py::TestLazyModules::test_lazy_batchnorm3d_pickle
[gw82] FAILED test/test_nn.py::TestLazyModules::test_lazy_instancenorm2d
[gw85] FAILED test/test_nn.py::TestLazyModules::test_lazy_batchnorm2d
~~~

**Comment by jansel** [comment]
Yes, I think this is something we should detect and support
